File: frontend/src/lib/emissionFactors.ts
// Emission factors — EXACT values from README spec
import type { TransportMode, DietType, ShoppingLevel } from '../types/carbon';

export const TRANSPORT_FACTORS: Record<TransportMode, number> = {
  car: 0.21,      // kg CO₂e per km
  bus: 0.089,
  train: 0.041,
  bike: 0,
  walk: 0,
};

export const DIET_MONTHLY_KG: Record<DietType, number> = {
  vegan: 55,
  vegetarian: 85,
  omnivore: 150,
  'meat-heavy': 230,
};

export const SHOPPING_MONTHLY_KG: Record<ShoppingLevel, number> = {
  low: 30,
  medium: 70,
  high: 130,
};

export const ELECTRICITY_FACTOR = 0.82; // kg CO₂e per kWh
export const FLIGHT_FACTOR = 0.255;     // kg CO₂e per km

// Eco Score logic — EXACT from README
export function calcEcoScore(monthlyKgCO2e: number): number {
  if (monthlyKgCO2e <= 250) return Math.round(90 + (250 - monthlyKgCO2e) / 250 * 10);
  if (monthlyKgCO2e <= 500) return Math.round(70 + (500 - monthlyKgCO2e) / 250 * 19);
  if (monthlyKgCO2e <= 800) return Math.round(50 + (800 - monthlyKgCO2e) / 300 * 19);
  if (monthlyKgCO2e <= 1200) return Math.round(30 + (1200 - monthlyKgCO2e) / 400 * 19);
  return Math.max(0, Math.round(29 - (monthlyKgCO2e - 1200) / 100));
}

export function getScoreMessage(monthlyKgCO2e: number): string {
  if (monthlyKgCO2e <= 250) return 'Excellent low-impact lifestyle';
  if (monthlyKgCO2e <= 500) return 'Good start with clear improvement options';
  if (monthlyKgCO2e <= 800) return 'Moderate footprint with useful action areas';
  if (monthlyKgCO2e <= 1200) return 'High footprint with major opportunities';
  return 'Very high footprint; focus on one category first';
}

File: backend/services/emission_factors.py
# Exact README emission factors

TRANSPORT_FACTORS = {
    "car": 0.21,        # kg CO2e/km
    "bus": 0.089,       # kg CO2e/km
    "train": 0.041,     # kg CO2e/km
    "bike": 0.0,
    "walk": 0.0
}

FLIGHT_FACTOR = 0.255   # kg CO2e/km

DIET_MONTHLY_KG = {
    "vegan": 55.0,
    "vegetarian": 85.0,
    "omnivore": 150.0,
    "meat-heavy": 230.0
}

ELECTRICITY_FACTOR = 0.82 # kg CO2e/kWh

SHOPPING_MONTHLY_KG = {
    "low": 30.0,
    "medium": 70.0,
    "high": 130.0
}
